### DISCUSSION TITLE
Typing encoding issue

### DISCUSSION THREAD

**Discussion by nvkhoi99** [openTCS/opentcs#14] [Q&A] [answered]
Hi, I'm now in research about AGV and get my customization from version 5.0 of openTCS

In progress, I detected that some character utf-8 I try to type to swing control is changed to "?" char
![image](https://user-images.githubusercontent.com/60660663/157913566-a5cfab5c-3e68-41b6-8641-4516c2394093.png)

After looked up in internet, I know that project is using windows-1252 encoding, but my efforts to fix and change to utf-8 not working.

How can I fix the problem. Thank so much for the open source and hope it will continue to be developed for a long time

**Comment by swltr** [comment, marked as answer]
You need to set the source file encoding in your editor/IDE. (Where exactly you have to do that depends on the editor you're using.)

If you just started using/customizing openTCS, please note that 5.0 is from June 2020. I recommend you start with the most recent version of openTCS, which is 5.4.

**Reply by nvkhoi99** [reply]
I'm using IntelliJ for coding, but sometime I also use NetBeans for swing UI designing
Tried to change project encoding but it's not really working
![image](https://user-images.githubusercontent.com/60660663/158023565-c5d29c1b-c6d4-4ada-8ffc-bf0004d2f657.png)

**Reply by swltr** [reply]
I cannot help with IntelliJ. It's possible that the IDE does not set/change the encoding of existing files, only for newly created ones.

For NetBeans, there is the "Encoding Support" plugin that lets you set the encoding for individual files, and I think you can set the default encoding via JVM properties if you modify NetBeans's startup script.
